src/Features/CSharp/Portable/ChangeSignature/CSharpChangeSignatureService.cs

```diff
@@ -286,6 +291,13 @@ private static bool InSymbolHeader(SyntaxNode matchingNode, int position)
                 return method.WithParameterList(method.ParameterList.WithParameters(updatedParameters).WithAdditionalAnnotations(changeSignatureFormattingAnnotation));
             }
 
+            // TODO: Record structs
+            if (updatedNode.IsKind(SyntaxKind.RecordDeclaration, out RecordDeclarationSyntax? record) && record.ParameterList is not null)
```

Comment:
just don't do a kind check, and do a syntax-check, and you'll be all good here.  
```suggestion
            if (updatedNode is RecordDeclarationSyntax { ParameterList: not null })
```

Reply:
![image](https://user-images.githubusercontent.com/31348972/116972816-e376af00-acbb-11eb-9a8d-988a8892d73c.png)

@CyrusNajmabadi Nullable analysis doesn't seem to handle patterns correctly.